examples/experimental/i3s-17-and-debug/components/tile-validator.js

```diff
@@ -88,18 +88,23 @@ export default class TileValidator extends PureComponent {
     super(props);
 
     this.state = {
+      tileId: props.tile.id,
       geometryInfo: null,
       triangleMessages: null,
       boundingVolumeInfo: null
     };
   }
 
-  componentDidUpdate(prevProps) {
-    const isNoTile = !this.props.tile || !prevProps.tile;
-
-    if (isNoTile || this.props.tile.id !== prevProps.tile.id) {
-      this.setState({geometryInfo: null, triangleMessages: null, boundingVolumeInfo: null});
+  static getDerivedStateFromProps(props, state) {
+    if (props.tile.id !== state.tileId) {
```

Comment:
Maybe a line of comment about what is happening here?

Reply:
Something like `// New tile is being set, re-initialize state`

Reply:
Added